R/Connection.R

```diff
@@ -608,24 +608,94 @@ setMethod(
 
 #' List Available ODBC Drivers
 #'
-#' List the available drivers on your system. See the
-#' [Installation](https://github.com/r-dbi/odbc#installation) section of the
-#' package README for details on how to install drivers for the most common
-#' databases.
-#'
-#' @param keep A character vector of driver names to keep in the results, if
-#'   `NULL` (the default) will keep all drivers.
-#' @param filter A character vector of driver names to filter from the results, if
-#'   `NULL` (the default) will not filter any drivers.
+#' Collect information about the available driver names.
+#'
+#' @param keep A character vector of driver names to keep in the results.
+#'   If `NULL`, all driver names will be kept.
+#'   Set the `odbc.drivers_keep` option to always keep a set of driver names.
+#' @param filter A character vector of driver names to remove from the results.
+#'   If `NULL`, no driver names will be removed.
+#'   Set the `odbc.drivers_remove` option to always remove a set of driver names.
+#'
 #' @return A data frame with three columns.
-#' If a given driver does not have any attributes the last two columns will be
-#' `NA`. Drivers can be excluded from being returned by setting the
-#' \code{odbc.drivers.filter} option.
+#'
 #' \describe{
-#'   \item{name}{Name of the driver}
-#'   \item{attribute}{Driver attribute name}
-#'   \item{value}{Driver attribute value}
+#'   \item{name}{Name of the driver. The entries in this column can be
+#'     passed to the `driver` argument of [dbConnect()] (as long as the
+#'     driver accepts the argument).}
+#'   \item{attribute}{Driver attribute name.}
+#'   \item{value}{Driver attribute value.}
 #' }
+#'
+#' @details
+#' Driver names are first processed with `keep`, then `filter`. Thus, if a
+#' driver name is in both `keep` and `filter`, it won't appear in output.
+#'
+#' If a driver has multiple attributes, there will be one row per attribute,
+#' each with the same driver `name`. If a given driver name does not have any
+#' attributes, the function will return one row with the driver `name`, but
+#' the last two columns will be `NA`.
+#'
+#' @section Configuration:
+#'
+#' This function interfaces with the driver manager to collect information
+#' about the available driver names.
+#'
+#' For **MacOS and Linux**, the odbc package supports the unixODBC driver
+#' manager. unixODBC looks to the `odbcinst.ini` _configuration file_ for
+#' information on driver names. Install unixODBC with `brew install unixodbc`,
+#' and find the location(s) of your `odbcinst.ini` file(s) with `odbcinst -j`.
```

Comment:
Related to https://github.com/r-dbi/odbc/issues/565#issuecomment-1781253881. I think some sort of `odbcLocateConfiguration()` may be nice here.

Reply:
In the meantime, I think you could remove the advice about how to install unixODBC since I think will have needed to do that well before you read these docs.